substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1302,11 +1306,18 @@ impl<T: Config> Pallet<T> {
                 - existential_deposit_requirement;
         }
 
-        <T as Config>::Currency::slash(&twin.account_id, amount_to_burn);
+        let to_burn = T::Currency::withdraw(
+            &twin.account_id,
+            amount_to_burn,
+            WithdrawReasons::TRANSACTION_PAYMENT,
```

Comment:
Transaction payment is meant for fees regarding actual transactions, which this isn't really. As such, WithdrawReasons::FEE would be a better fit I think